recipes/cppbenchmark/all/CMakeLists.txt

```
@@ -0,0 +1,37 @@
+cmake_minimum_required(VERSION 3.13)
+project(cmake_wrapper)
+
+include(${CMAKE_CURRENT_BINARY_DIR}/../conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(WIN32)
+    add_definitions (-D_CRT_SECURE_NO_WARNINGS)
+endif()
+
+enable_testing()
+
+add_subdirectory("source_subfolder")
+
+conan_target_link_libraries(cppbenchmark)
+
+target_compile_features(cppbenchmark PUBLIC cxx_std_17)
```

```suggestion
target_compile_features(cppbenchmark PUBLIC cxx_std_17)
```

It could be sent to the upstream.

---

The project handles this through its own package management. It makes a reference to another repo in the `.gitlinks` file (https://github.com/chronoxor/CppBuildScripts.git) and sets `CMAKE_CXX_STANDARD` to 17 in a cmake script that lives in this other repo:

```cmake
# File: SetCompilerFeatures.cmake
# Compiler features

# Choose C++ standard
set(CMAKE_CXX_STANDARD 17)
```

So, as conan already handles this kind of thing I've removed any handling made through the project own dependency and compile features management and used this instead:

https://github.com/conan-io/conan-center-index/blob/7db23f01d1c9388a55a3c4e4f3fc041ecddc83fe/recipes/cppbenchmark/all/conanfile.py#L32-L39

https://github.com/conan-io/conan-center-index/blob/7db23f01d1c9388a55a3c4e4f3fc041ecddc83fe/recipes/cppbenchmark/all/conanfile.py#L61-L73

But this is not enough to force C++ 17 mode and so I added it with the compiler feature thing.

---

@tarc it's the same problems we saw with #3224, sadly upstream does not have the most consumer friendly cmake...

This _was_ considered a back practice in CMake for the longest time but it's been permissible for a while.

As a rule of thumb for CCI, would we be happy with the or rather have it in a patch?

---

No problem, because by default cppstd is None and we can't force a specific versions, otherwise we would break all users. So, if a project requires a minimal C++ standard, but it uses another way to set it, we can force it directly by the cmake wrapper. We have more recipes using this, it's not a common case, but happens.

---

Thanks for confirming!